oat-sa/expr-eval
# Silentmatt v2.0.2

### Related to: https://oat-sa.atlassian.net/browse/TR-5072

### Summary

Add changes from version `2.0.2` of the base repository.

Also, fix a security issue by replace Uglify with Terser

### Details

Version `2.0.2` is the last version from the base repo. A few months ago, we merged the previous major version as the changes from the 2 looked too big. Finally, here is the merge for the latest version, with resolved conflict.

Actually, the code does not need to be scrutinized since this is mostly a merge from the base repo. What needs to be checked is if the tests pass (both in this lib and in the calculator's engine).

### How to test
- install the package, and see that no vulnerabilities are found: `npm i`
- run the unit tests: `npm test`
- install https://github.com/oat-sa/tao-calculator-fe, then clone the pull request in place of `node_modules/@oat-sa/expr-eval`:
    ```
    cd node_modules/@oat-sa/
    rm -rf expr-eval
    git clone git@github.com:oat-sa/expr-eval.git -b silentmatt-v2.0.2
    cd expr-eval
    npm i
    cd ../../..
    npm test
    ```
